lib/lightning/projects.ex

```diff
@@ -697,4 +698,125 @@ defmodule Lightning.Projects do
   defp delete_history_for(_project) do
     {:error, :missing_history_retention_period}
   end
+
+  def invite_collaborators(project, collaborators, inviter) do
+    Multi.new()
+    |> Multi.put(:collaborators, collaborators)
+    |> Multi.merge(&register_users/1)
+    |> Multi.run(:add_users_to_project, fn _repo, changes ->
+      add_users_to_project(changes, project, collaborators)
+    end)
+    |> Multi.run(:send_invitations, fn _repo, changes ->
+      send_invitations(changes, project, inviter)
+    end)
+    |> execute_transaction()
+  end
+
+  defp execute_transaction(%Ecto.Multi{} = multi) do
```

Comment:
This is a bit picky but `Repo.transaction` already has a clear semantics of executing a transaction. Maybe we needed this in multiple places but now it is used only in one so it would better to remove the indirection.